Add example for dynamically mapped tasks

potiuk opened the pull request:
<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

josh-fell commented on airflow/example_dags/example_mapped_task.py:
```diff
@@ -0,0 +1,66 @@
+
+from airflow import DAG, XComArg
+from airflow.decorators import task
+from airflow.operators.bash_operator import BashOperator
```
```suggestion
from airflow.operators.bash import BashOperator
```

potiuk replied:
Aha. Right.

josh-fell commented on airflow/example_dags/example_mapped_task.py:
```diff
@@ -0,0 +1,66 @@
+
+from airflow import DAG, XComArg
+from airflow.decorators import task
+from airflow.operators.bash_operator import BashOperator
```
```suggestion
from airflow.operators.bash import BashOperator
```

josh-fell reviewed (commented):
Otherwise LGTM

ephraimbuddy commented on airflow/example_dags/example_mapped_task.py:
```diff
@@ -0,0 +1,66 @@
+
+    @task
+    def add_one(x: int, multiplier: int = 1):
+        return (x + 1) * 2
```
```suggestion
        return (x + multiplier) * 2
```

jedcunningham replied:
Good call, but shouldn't it be:
```suggestion
        return (x + 1) * multiplier
```

potiuk replied:
Right. I focused on getting it complex and try out all the features of dymamic tasks, not to be correct :)

potiuk commented on airflow/example_dags/example_mapped_task.py:
```diff
@@ -0,0 +1,66 @@
+    bash_input = echo_task.expand(value=third)
+    sum_it(third)
+
+    classic = BashOperator.partial(task_id="final").expand(bash_command=bash_input)
```
```suggestion
    classic = BashOperator.partial(task_id="classic").expand(bash_command=bash_input)
```

potiuk committed:
Add airflow/example_dags/example_mapped_task.py

This change adds quite a bit of modifications in the mapped operators
implementation to make them comparable before and after serialization
as otherwise the serialization tests fail.

pingzh reviewed (approved):
nice, thanks
